jobs/build/reposync/Jenkinsfile

```diff
@@ -40,38 +48,62 @@ node {
 
     MIRROR_TARGET = "use-mirror-upload.ops.rhcloud.com"
-    MIRROR_PATH = "/srv/enterprise/reposync/${SYNC_VERSION}/"
+    MIRROR_REPOSYNC_BASE_DIR = "/srv/enterprise/reposync/"
 
     // doozer_working must be in WORKSPACE in order to have artifacts archived
     DOOZER_WORKING = "${WORKSPACE}/doozer_working"
     buildlib.cleanWorkdir(DOOZER_WORKING)
 
     try {
-        sshagent(['openshift-bot']) {
-            // To work on real repos, buildlib operations must run with the permissions of openshift-bot
+        lock('buildvm-yum') { // prevent simultaneous job runs from conflicting over yum lock
+            sshagent(['openshift-bot']) {
+                // To work on real repos, buildlib operations must run with the permissions of openshift-bot
+
+                stage("sync repos to local") {
+                    cacheDir = "${LOCAL_CACHE_DIR}_${ARCH}"
+
+                    if ( ARCH == 'x86_64' ) {
+                        // Match legacy location for x86_64
+                        syncDir = "${LOCAL_SYNC_DIR}"
+                    } else {
+                        // Non x86_64 arch directories will have the arch as a suffix
+                        syncDir = "${LOCAL_SYNC_DIR}_${ARCH}"
+                    }
 
-            stage("sync repos to local") {
-                command = "--working-dir ${DOOZER_WORKING} --group 'openshift-${SYNC_VERSION}' "
-                command += "beta:reposync --output ${LOCAL_SYNC_DIR} --cachedir ${LOCAL_CACHE_DIR} --repo-type ${REPO_TYPE} "
-		try {
+                    command = "--working-dir ${DOOZER_WORKING} --group 'openshift-${SYNC_VERSION}' "
+                    command += "beta:reposync --output ${syncDir}/ --cachedir ${cacheDir}/ --repo-type ${REPO_TYPE} --arch ${ARCH}"
                     buildlib.doozer command
-		} catch (err) {
-		    echo "whoops. fix this tomorrow please"
-		}
-            }
 
-            stage("push to mirror") {
-                sh "rsync -avzh --delete -e \"ssh -o StrictHostKeyChecking=no\" ${LOCAL_SYNC_DIR} ${MIRROR_TARGET}:${MIRROR_PATH} "
-                mirror_result = buildlib.invoke_on_use_mirror("push.enterprise.sh")
-                if (mirror_result.contains("[FAILURE]")) {
-                    echo mirror_result
-                    error("Error running ${SYNC_VERSION} reposync push.enterprise.sh:\n${mirror_result}")
+                    /**
+                     * A bug in doozer once caused only noarch RPMs to be synced to the mirrors. Prevent this
+                     * by making sure a minimum number of arch-specific RPMs actually exist locally before
+                     * pushing the updated content to the mirrors.
+                     */
+
+                    sanityCheckRes = commonlib.shell(
+                            returnAll: true,
+                            script: "find ${syncDir} -name '*.${ARCH}.rpm' | wc -l"
+                    )
+
+                    if(sanityCheckRes.stdout.trim().toInteger() <  10000){
```

Comment:
10000 seems slightly excessive...

Reply:
The actual number was 14000, so I was being conservative :) . Glad to set to whatever you want.

Reply:
That's assuming we continue with most of the ones we have now. But that's reasonable for now.